### PR TITLE
Make Vect follow the naming convention of List

### PR DESCRIPTION
**Pull request by david-christiansen**
Rename Vect n a to Vect len elem, where it makes sense.

### CODE REVIEW

File: libs/base/Data/Vect.idr [hunk 1]
```diff
@@ -9,13 +9,15 @@ import Language.Reflection
...
 %elim
-data Vect : Nat -> Type -> Type where
+data Vect : (len : Nat) -> (elem : Type) -> Type where
   ||| Empty vector
-  Nil  : Vect Z a
+  Nil  : Vect Z elem
   ||| A non-empty vector of length `S k`, consisting of a head element and
   ||| the rest of the list, of length `k`.
-  (::) : (x : a) -> (xs : Vect k a) -> Vect (S k) a
+  (::) : (x : elem) -> (xs : Vect k elem) -> Vect (S k) elem
```
Comment (Melvar) [inline libs/base/Data/Vect.idr]: This feels like it is conspicuously left out of the conversion to `len`.

File: libs/base/Data/Vect.idr [hunk 2]
```diff
@@ -283,7 +286,7 @@ implementation Functor (Vect n) where
 |||
 ||| @ f the partial function (expressed by returning `Maybe`)
 ||| @ xs the vector to check for results
-mapMaybe : (f : a -> Maybe b) -> (xs : Vect n a) -> (m : Nat ** Vect m b)
+mapMaybe : (f : elem -> Maybe b) -> (xs : Vect len elem) -> (m : Nat ** Vect m b)
```
Comment (Melvar) [inline libs/base/Data/Vect.idr]: The use of `elem` and `b` as opposed to `a` and `b` seems odd to me, given they’re both the element type of different `Vect`s.

**Comment by david-christiansen** [comment]
Thanks @Melvar !

**Comment by Melvar** [comment]
You’re welcome, just noting things that seemed out of place.

**Comment by jfdm** [comment]
@david-christiansen Looks like a few of the expected files in the test suite were not updated.

**Comment by david-christiansen** [comment]
@jfdm yeah, I'll update it when I'm on a faster computer.

**Comment by jfdm** [comment]
@david-christiansen I took the liberty to fix the changes for you and run the tests locally. Just so you are aware, I amended the commit myself so when i merged your commit manually from the command line, this PR was not automagically closed.

**Comment by Melvar** [comment]
For the future, [Improving collaboration with forks](https://github.com/blog/2247-improving-collaboration-with-forks) shows how to set things up to make such changes on the same branch so that GitHub knows about the merge.

**Comment by david-christiansen** [comment]
@jfdm thanks for taking care of this for me!

**Comment by jfdm** [comment]
@david-christiansen No worries.

@Melvar I was aware of the improved collaboration features, but as the PR was created before they were rolled out I could not do that with this PR.

**Pull request closed without merging**